lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -505,6 +506,15 @@ def query(table_name, opts = {})
           record_count = 0
           scan_count = 0
           loop do
+            # Adjust the limit down if the remaining record or scan limit are
+            # lower to obey limits. We can assume the difference won't be
+            # negative due to break statements below.
+            if q[:limit] && record_limit && record_limit - record_count < q[:limit]
+              q[:limit] = record_limit - record_count
+            elsif q[:limit] && scan_limit && scan_limit - scan_count < q[:limit]
+              q[:limit] = scan_limit - scan_count
+            end
+
```

Comment:
I beleive we should honor `scan_limit` as well as `record_limit` (it was described [here ](https://github.com/Dynamoid/Dynamoid/pull/172#discussion_r133783235)). So I can imagine case when `record_limit - record_count` is greater than `scan_limit - scan_count`. That's why I think both conditions should be separated. I mean

```ruby
            if q[:limit] && record_limit && record_limit - record_count < q[:limit]
               q[:limit] = record_limit - record_count
            end
            if q[:limit] && scan_limit && scan_limit - scan_count < q[:limit]
              q[:limit] = scan_limit - scan_count
            end

```